packages/design-system/src/components/Modal/Modal.tsx

```diff
@@ -1,88 +1,121 @@
-import React from 'react';
-import { useDialogState, DialogProps, Dialog as ReakitDialog } from 'reakit';
-import { useTranslation } from 'react-i18next';
+import React, { ReactElement, ReactNode } from 'react';
+import i18n from 'i18next';
+import { Dialog, DialogBackdrop } from 'reakit/Dialog';
+import { IconName } from '@talend/icons';
 
-import { ButtonSecondary, ButtonPrimary } from '../Button';
+import { ButtonDestructive, ButtonPrimary, ButtonSecondary } from '../Button';
 import { Icon } from '../Icon';
+import { StackHorizontal, StackVertical } from '../Stack';
+import { ButtonPrimaryPropsType } from '../Button/variations/ButtonPrimary';
+import { ButtonSecondaryPropsType } from '../Button/variations/ButtonSecondary';
 
-import * as S from './Modal.style';
+import modalStyles from './Modal.scss';
 
-export type ModalProps = DialogProps & {
-	icon?: string;
-	disclosure: React.ComponentPropsWithRef<any>;
-	subtitle?: string;
-	onClose?: () => void;
-	onValidate?: () => void;
+type ModalIcon = IconName | ReactElement;
+
+function ModalIcon(props: { icon: ModalIcon; 'data-test'?: string }): ReactElement {
+	const { icon, ['data-test']: dataTest } = props;
+	return (
+		<div className={modalStyles['modal-icon']} data-test={dataTest}>
+			{typeof icon === 'string' ? <Icon name={icon} /> : icon}
+		</div>
+	);
+}
+
+export type ModalPropsType = {
+	header: {
+		title: ReactNode;
+		description?: string;
+		icon?: ModalIcon;
+	};
+	onClose: Function;
+	primaryAction?: ButtonPrimaryPropsType & { destructive?: boolean };
+	secondaryAction?: ButtonSecondaryPropsType;
+	preventEscaping?: boolean;
+	children: ReactNode | ReactNode[];
 };
 
-const Modal = React.forwardRef(
-	(
-		{ disclosure, children, icon, title, subtitle, onClose, onValidate, ...props }: ModalProps,
-		ref: React.Ref<HTMLDivElement>,
-	) => {
-		const dialog = useDialogState({ animated: true });
-		const { t } = useTranslation();
+function Modal(props: ModalPropsType): ReactElement {
+	const { header, onClose, primaryAction, secondaryAction, preventEscaping, children } = props;
+	const dialog = { visible: true };
```

Comment:
```suggestion
      const dialog = useDialogState({ visible: true });
```
For now. But we can't skip the Reakit state management.

Reply:
Let me have a look so that I don't leave traps in the code for our future selves